src/components/CreateAuthor.jsx

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useState } from 'react';
+import { Deserializer } from 'jsonapi-serializer';
+import useAuth from '../hooks/useAuth';
+
+const initialValues = {
+  firstName: '',
+  lastName: '',
+  birthDate: '',
+};
+
+export default function CreateAuthor({ setAuthors }) {
+  const [values, setValues] = useState(initialValues);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const { currentUser } = useAuth();
+
+  const handleSubmit = async function handleSubmit(event) {
+    event.preventDefault();
+    const requestOptions = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${currentUser?.access_token}`,
+      },
+      body: JSON.stringify({ ...values }),
+    };
+    fetch(`${process.env.REACT_APP_API_URL}/api/authors`, requestOptions)
+      .then((response) => {
+        if (!response.ok) {
+          setError(true);
+          return {};
+        }
+        return response.json();
+      })
+      .then((data) => {
+        new Deserializer({ keyForAttribute: 'camelCase' })
+          .deserialize(data, (_error, author) => setAuthors((prevState) => [...prevState, author]));
```

Comment:
Aquí podría ser mejor pasar como prop una función tipo `addAuthor`, que reciba como parámetro un author deserializado, entonces sólo se llamaría a `addAuthor(author)`:
```suggestion
          .deserialize(data, (_error, author) => addAuthor(author));
```

Luego en `<AuthorList />` defines esa función con lo que agregaste aquí:
```javascript
const addAuthor = (author) => setAuthors((prevState) => [...prevState, author]);
```
La razón es que `CreateAuthor` no debería saber del estado del padre, y al utilizar `setAuthors` de esta manera se está suponiendo que este componente sabe cómo actualizar el estado del padre, lo que genera un acomplamiento entre ambos.

Reply:
Toda la razón. Ya lo corregí.